django/db/models/query.py

```diff
@@ -521,13 +520,12 @@ def _create_object_from_params(self, lookup, params):
                 params = {k: v() if callable(v) else v for k, v in params.items()}
                 obj = self.create(**params)
             return obj, True
-        except IntegrityError:
-            exc_info = sys.exc_info()
+        except IntegrityError as e:
             try:
                 return self.get(**lookup), False
             except self.model.DoesNotExist:
                 pass
-            raise exc_info[0](exc_info[1]).with_traceback(exc_info[2])
+            raise e
```

Comment:
I don't think the explicit `raise e` is required on Python 3.5+.
```python
(tmp-70aba34da671918e) ➜  tmp-70aba34da671918e ipython
Python 3.5.2 (default, Sep 17 2016, 14:04:19)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: def func():
   ...:     try:
   ...:         raise KeyError('foo')
   ...:     except KeyError:
   ...:         try:
   ...:             raise Exception('test')
   ...:         except Exception:
   ...:             pass
   ...:         raise
   ...:

In [2]: func()
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-bd1982955a12> in <module>()
----> 1 func()

<ipython-input-1-e80913097bf5> in func()
      1 def func():
      2     try:
----> 3         raise KeyError('foo')
      4     except KeyError:
      5         try:

KeyError: 'foo'
```

Reply:
Yes, I remember `raise` vs. `raise e` being a point of confusion in the past. If it's just `raise`, then lines 3-4 below are omitted in the traceback:
```
File "/home/tim/code/django/django/db/models/query.py", line 489, in get_or_create
  return self._create_object_from_params(lookup, params)
File "/home/tim/code/django/django/db/models/query.py", line 528, in _create_object_from_params
  raise e
File "/home/tim/code/django/django/db/models/query.py", line 521, in _create_object_from_params
  obj = self.create(**params)
```
It could possibly be unclear without those lines, but I'm not sure if it matters.

Reply:
Makes sense, let's stick with `raise e`.